### PR TITLE
adding Colorado medicare exclusion list crawler

### PR DESCRIPTION
**Pull request by bgmello**

### CODE REVIEW

**Review by jbothma — CHANGES_REQUESTED** [review]
Thank you!

Same changes as kansas please

**Comment by bgmello** [comment]
I'm getting an error while trying to create the link between the company and the entity saying it's a self referencing link. I left that part of the code commented out.

File: datasets/us/co/med_exclusions/crawler.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    entity = context.make("LegalEntity")
+    entity.id = context.make_id(name, row.get("npi"))
+    entity.add("name", name)
+
+    if row.get("npi") != "N/A":
+        npis = re.split(r"; |& ", row.pop("npi"))
+
+        for npi in npis:
+            entity.add("npiCode", npi)
```
Comment (jbothma) [inline datasets/us/co/med_exclusions/crawler.py:25]: entity.add happily takes lists

```suggestion
        entity.add("npiCode", npis)
```

File: datasets/us/co/med_exclusions/crawler.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    if not name:
+        return
+
+    entity = context.make("LegalEntity")
+    entity.id = context.make_id(name, row.get("npi"))
+    entity.add("name", name)
+
+    if row.get("npi") != "N/A":
+        npis = re.split(r"; |& ", row.pop("npi"))
```
Comment (jbothma) [inline datasets/us/co/med_exclusions/crawler.py:22]: no need to change, but h.multisplit() is a common tool for this purpose

File: datasets/us/co/med_exclusions/crawler.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    sanction = h.make_sanction(context, entity)
+    termination_effective_date = row.pop("termination_effective_date")
+    sanction.add("startDate", termination_effective_date)
+    sanction.add("reason", row.pop("termination_authority"))
+
+    if "reinstatement_effective_date" in row:
+        reinstatement_effective_date = row.pop("reinstatement_effective_date")
+    else:
+        reinstatement_effective_date = None
```
Comment (jbothma) [inline datasets/us/co/med_exclusions/crawler.py:37]: or simply

```suggestion
    reinstatement_date = row.pop("reinstatement_effective_date", None)
```

File: datasets/us/co/med_exclusions/us_co_med_exclusions.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  start: "2024-08-15"
+load_db_uri: ${OPENSANCTIONS_DATABASE_URI}
+summary: >
+  This dataset contains a list of providers who have been terminated for cause from Colorado's Medicaid program.
+description: |
+  The Department of Health Care Policy & Financing (HCPF) maintains a list of providers whose participation in
+  Health First Colorado (Colorado's Medicaid program) has been terminated for cause. A provider who has been
+  terminated for cause may not receive reimbursement under the Medicaid program. A description of the reasons
+  a provider may be terminated for cause can be found under the definition of Good Cause at 10 CCR 2505-10,
+  Section 8.076.1.7.
```
Comment (jbothma) [inline datasets/us/co/med_exclusions/us_co_med_exclusions.yaml:15]: If we quote verbatim, it's best to indicate that with markdown quote prefixes and a link

```suggestion
  > The Department of Health Care Policy & Financing (HCPF) maintains a list of providers whose participation in
  > Health First Colorado (Colorado's Medicaid program) has been terminated for cause. A provider who has been
  > terminated for cause may not receive reimbursement under the Medicaid program. A description of the reasons
  > a provider may be terminated for cause can be found under the definition of Good Cause at 10 CCR 2505-10,
  > Section 8.076.1.7.
  
  [Provider Termination page](https://hcpf.colorado.gov/provider-termination)
```

File: datasets/us/co/med_exclusions/crawler.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    dba = row.pop("doing_business_as_name")
+    if dba:
+        company = context.make("Company")
+        company.id = context.make_id(dba)
+
+        if target:
+            company.add("topics", "debarment")
+
+        # link = context.make("UnknownLink")
+        # link.id = context.make_id(dba)
```
Comment (jbothma) [inline datasets/us/co/med_exclusions/crawler.py:60]: the link ID and company ID would end up the same unless you add something to the link

```suggestion
        link.id = context.make_id(entity.id, company.id)
```

**Pull request merged**